src/Microsoft.VisualStudio.Threading.Analyzers.Tests/VSTHRD200UseAsyncNamingConventionAnalyzerTests.cs

```diff
@@ -35,6 +35,54 @@ class Test {
             await Verify.VerifyCodeFixAsync(test, expected, withFix);
         }
 
+        [Fact]
+        public async Task IVsTaskReturningMethodWithoutSuffix_WithGetAwaiter_GeneratesWarning()
+        {
+            var test = @"
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+class Test {
+    IVsTask T() => null;
+}
+";
+
+            var withFix = @"
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+class Test {
+    IVsTask TAsync() => null;
+}
+";
+
+            var expected = Verify.Diagnostic(AddSuffixDescriptor).WithSpan(6, 13, 6, 14);
+            await Verify.VerifyCodeFixAsync(test, expected, withFix);
+        }
+
+        [Fact]
+        public async Task IVsTaskReturningMethodWithSuffix_NoGetAwaiter_GeneratesWarning()
+        {
+            var test = @"
+using Microsoft.VisualStudio.Shell.Interop;
+
+class Test {
+    IVsTask TAsync() => null;
```

Comment:
If I understand this correctly, I am not sure how I feel about this. While it is true code doesn't have access to GetAwaiter extension method due to usings, we know the type is awaitable. Granted this will likely not happen much in real life since most code using IVsTask already has a using for MS.VS.Shell but I am wondering if we should hard code some types?

Reply:
I'm going to have to fix this anyway... the cache is wrong: it caches at the Compilation level information that is specific to a position within a document. That's no good. So I'll have to do a compilation-wide cache (or broader) which means the `using` directives won't matter. :)

Reply:
Fixed. But now we need to come up with a better caching story since the performance of this is abysmal. I'm hoping @sharwell will have some good ideas.